### PR TITLE
Add Solana block leaders utils

### PR DESCRIPTION
**Pull request by 0xBoxer**
This PR adds the Solana block leaders utils.

### CODE REVIEW

**Comment by 0xBoxer** [comment]
@jeff-dude @0xRobin can I get a merge on this, need this to be merged to progress on [this PR ](https://github.com/duneanalytics/spellbook/pull/6926)as editing 2 subprojects at once doesn't seem to work - runtimes seem alright and this model is completely isolated.

File: dbt_subprojects/solana/models/solana_utils/solana_utils_block_leaders.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+FROM {{ source('solana', 'blocks') }} b 
+LEFT JOIN {{ source('solana', 'rewards') }} r 
+    ON b.slot = r.block_slot 
+    AND r.reward_type = 'Fee'
+
+--not all slots have leaders, but slots without leaders don't have transactions so can be disregarded.
+
+{% if is_incremental() %}
+    AND {{incremental_predicate('b.time')}}
+{% endif %}
```
Comment (0xRobin) [inline dbt_subprojects/solana/models/solana_utils/solana_utils_block_leaders.sql:42]:
```suggestion
FROM {{ source('solana', 'blocks') }} b 
LEFT JOIN {{ source('solana', 'rewards') }} r 
    ON b.slot = r.block_slot 
    AND r.reward_type = 'Fee'
    {% if is_incremental() %}
    AND {{incremental_predicate('r.block_time')}}
    {% endif %}
{% if is_incremental() %}
WHERE {{incremental_predicate('b.time')}}
{% endif %}
```

If we can drop the blocks that don't have a leader, we can make this an INNER join and move both the `incremental_predicates` into the join condition.

Comment (0xRobin) [reply dbt_subprojects/solana/models/solana_utils/solana_utils_block_leaders.sql:42]: FYI, I noticed the incremental run was taking longer then the initial run in CI, which is a good sign that's something's up

File: dbt_subprojects/solana/models/solana_utils/_schema.yml [hunk 2]
```diff
@@ -105,4 +105,48 @@ models:
...
\ No newline at end of file
+        name: owned_domains
+
+  - name: solana_utils_block_leaders
+    meta:
+      blockchain: solana
+      contributors: [0xBoxer]
+    config:
+      tags: ['solana','block_leaders']
+    description: >
```
[the remark this answers is not in the document]
Comment (0xBoxer) [reply dbt_subprojects/solana/models/solana_utils/_schema.yml:116]: done in https://github.com/duneanalytics/spellbook/pull/6931/commits/553024ec1c64f512001a7bc55cdad55a5fd09aa6

**Review by jeff-dude — COMMENTED** [review]
nice optimization, blazing fast ⚡

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**